server/schedulers/balance_region.go

```diff
@@ -191,21 +189,21 @@ func (s *balanceRegionScheduler) Schedule(cluster schedule.Cluster, dryRun bool)
 			// Priority pick the region that has a pending peer.
 			// Pending region may mean the disk is overload, remove the pending region firstly.
 			solver.region = filter.SelectOneRegion(cluster.RandPendingRegions(solver.SourceStoreID(), s.conf.Ranges), collector,
-				baseRegionFilters...)
+				append(baseRegionFilters, filter.NewRegionWitnessFilter(solver.SourceStoreID()))...)
```

Comment:
how about add `NewRegionWitnessFilter` to base region filters .

Reply:
I don't think it can, because `solver.SourceStoreID()` needs to be passed as a parameter, and the solver is obtained by traversing `sourceStores`.